torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -889,6 +889,12 @@ class FullyShardedDataParallel(nn.Module):
             This can also help load checkpoints taken by ``state_dict`` and to be loaded by
             ``load_state_dict`` in a memory efficient way. See documentation for
             :class:`FullStateDictConfig` for an example of this. (Default: ``False``)
+        all_gather_issue_limit (Optional[int]): If this is not ``None`` and
+            the sharding strategy is ``FULL_SHARD``, this represents the
+            maximum number of actively issued all-gathers. When this limit is
+            reached, FSDP blocks the CPU thread to ensure that some FSDP
+            parameters are freed before issuing further all-gathers. (Default:
```

Comment:
mention what this can help (gpu memory) and how to configure it?

Reply:
I am still working on getting a complete understanding. I will update the docstring once I do!